stat/stat.go

```diff
@@ -217,6 +217,35 @@ func Correlation(x, y, weights []float64) float64 {
 	return (sxy - xcompensation*ycompensation/sumWeights) / math.Sqrt(sxx*syy)
 }
 
+// KendallCorrelation returns the weighted Kendall correlation between the
+// samples of x and y.
+// The lengths of x and y must be equal.
+func KendallCorrelation(x, y []float64) float64 {
+	if len(x) != len(y) {
+		panic("stat: slice length mismatch")
+	}
+
```

Comment:
Don't we need to verify that `x` is sorted? Otherwise this test returns meaningless results I think.